### PR TITLE
Adds documentation for creating dynamic infrastructure for step package targets

### PR DESCRIPTION
**Pull request by andrewabest**
In August we added functionality to dynamically create targets emitted from step packages.

This PR documents the new functions.

Related changes:

https://github.com/OctopusDeploy/Calamari/pull/785
https://github.com/OctopusDeploy/OctopusDeploy/pull/9774
https://github.com/OctopusDeploy/Calamari/commit/2413e18f6f6fb409f2c58dbe3ada53b339bd199f

### CODE REVIEW

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md [hunk 1]
```diff
@@ -50,11 +48,19 @@ Only a subset of account types and deployment targets support being created dyna
...
+
+Targets defined by step packages use the new generic `New-OctopusTarget` function to create targets. 
+:::
+
 - [Azure Web App](/docs/infrastructure/deployment-targets/dynamic-infrastructure/azure-web-app-target.md)
 - [Azure Service Fabric](/docs/infrastructure/deployment-targets/dynamic-infrastructure/azure-service-fabric-target.md)
 - [Azure Cloud Service](/docs/infrastructure/deployment-targets/dynamic-infrastructure/azure-cloud-service-target.md)
 - [Kubernetes Cluster](/docs/infrastructure/deployment-targets/dynamic-infrastructure/kubernetes-target.md)
-- [Remove Target](/docs/infrastructure/deployment-targets/dynamic-infrastructure/remove-target.md)
+- [AWS ECS Cluster](/docs/infrastructure/deployment-targets/dynamic-infrastructure/new-octopustarget.md)
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:62]:
```suggestion
- [New Target](/docs/infrastructure/deployment-targets/dynamic-infrastructure/new-octopustarget.md)
```

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md [hunk 2]
```diff
@@ -50,11 +48,19 @@ Only a subset of account types and deployment targets support being created dyna
 Before you can create dynamic targets in an Environment, the environment needs to be configured to allow it. See [Enabling dynamic infrastructure](/docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md#enable-dynamic-infrastructure) for more information.
 :::
 
+:::hint
+**Targets from Step Packages**
+Octopus Deploy has recently developed a new architecture for deployment steps and targets, known as step packages. To learn more about step packages, read the [step package documentation](https://github.com/octopusdeploy/step-api#overview).
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:53]: Does this need to include the version these became available (2021.2 / 3?)

**Review by harrisonmeister — COMMENTED** [review]
LGTM. Made one comment about calling out the version of Octopus this is available in (https://github.com/OctopusDeploy/OctopusDeploy/pull/9774) 2021.3

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md (hunk 1 above)
Comment (andrewabest) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md]: I've deliberately left this as the ECS cluster here taking them to New Target so people will more easily find what they are looking for. I think once we don't have "two ways" of doing this we can consolidate and simplify the list, but for now, I'd like to keep the targets listed separately. All new targets will just link to the same new target doc.

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md (hunk 2 above)
Comment (andrewabest) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:53]: That is a good idea, I'll add that.

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md (hunk 1 above)
Comment (harrisonmeister) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md]: Ah sorry I thought it was a typo 😃. I might have committed the change. I can revert tomorrow if you don't get chance to

Comment (andrewabest) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md]: All good I'm already in there, I'll fix it up

Comment (harrisonmeister) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md]: Thanks!

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md [hunk 3]
```diff
@@ -50,11 +48,19 @@ Only a subset of account types and deployment targets support being created dyna
 Before you can create dynamic targets in an Environment, the environment needs to be configured to allow it. See [Enabling dynamic infrastructure](/docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md#enable-dynamic-infrastructure) for more information.
 :::
 
+:::hint
+**Targets from Step Packages**
+Octopus Deploy has recently developed a new architecture for deployment steps and targets, known as step packages. Step packages are available in Octopus Deploy version 2021.3 and later. To learn more about step packages, read the [step package documentation](https://github.com/octopusdeploy/step-api#overview).
```
Comment (johnsimons) [inline docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:53]: @andrewabest how would a user identify a step package vs other steps?

Comment (andrewabest) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:53]: At the moment there isn't a good way to do so aside from reading the docs - in the `new-octopustarget` document I mention that not all targets are from step packages, and the ones that are are in that doc. I'm going to think about this more along with our "first class API experience" work as I feel it all ties together a bit.

Comment (johnsimons) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:53]: it makes me wonder if this "hint" even need to be here?
it feels a bit out of context, this is about creating dynamic targets

Comment (andrewabest) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:53]: You're probably right, it is a little out of context. The same hint is available in the other document, so I'll remove it from this one.

Comment (andrewabest) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md]: Done

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/new-octopustarget.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+```powershell PowerShell
+$inputs = @"
+{
+    "clusterName": "$($OctopusParameters["clusterName"])",
+    "region": "$($OctopusParameters["region"])",
+    "awsAccount": "$($OctopusParameters["awsAccount"])",
+}
+"@
+
+New-OctopusTarget -Name $OctopusParameters["target_name"] -TargetId "aws-ecs-target" -Inputs $inputs -Roles $OctopusParameters["role"]
```
Comment (johnsimons) [inline docs/infrastructure/deployment-targets/dynamic-infrastructure/new-octopustarget.md:59]: target names and roles can have spaces, I think it is better to quote these args

Comment (andrewabest) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/new-octopustarget.md]: Done

**Pull request merged**